packages/arb-token-bridge-ui/src/token-bridge-sdk/Erc20WithdrawalStarter.ts

```diff
@@ -94,38 +103,45 @@ export class Erc20WithdrawalStarter extends BridgeTransferStarter {
       throw Error('Erc20 token address not found')
     }
 
-    const destinationChainErc20Address =
-      await this.getDestinationChainErc20Address()
-
     const address = await getAddressFromSigner(signer)
 
     const sourceChainId = await getChainIdFromProvider(this.sourceChainProvider)
 
-    const destinationChainId = await getChainIdFromProvider(
-      this.destinationChainProvider
-    )
+    const gatewayAddress = await this.getSourceChainGatewayAddress()
+
+    const sourceChainTokenBridge = getArbitrumNetwork(sourceChainId).tokenBridge
 
-    // check first if token is even eligible for allowance check on l2
-    if (
-      (await tokenRequiresApprovalOnL2({
-        tokenAddressOnParentChain: destinationChainErc20Address,
-        parentChainId: destinationChainId,
-        childChainId: sourceChainId
-      })) &&
-      this.sourceChainErc20Address
-    ) {
-      const token = ERC20__factory.connect(
-        this.sourceChainErc20Address,
-        this.sourceChainProvider
-      )
-
-      const gatewayAddress = await this.getSourceChainGatewayAddress()
-      const allowance = await token.allowance(address, gatewayAddress)
-
-      return allowance.lt(amount)
+    // tokens that use the standard gateways do not require approval on child chain
+    const standardGateways = [
```

Comment:
Can we simply check if the token contract has a valid and working `bridgeBurn` method to validate the same without needing to fetch the gateway address and checking standardGatways and introducing `isError` in gas estimation?

Reply:
we need the gateway address for getting `allowanceForSourceChainGateway` anyway

we can do an `eth_call` (`provider.call`) to check if the `bridgeBurn` method reverts with a message (means the method exists), but if the token contract implemented the method to revert without a message, then we might wrongly think that the method doesn't exist when it does